main.py

```diff
@@ -214,13 +215,16 @@ def main(request) -> typing.Tuple[str, int]:
         def publish(records: typing.List[dict], observe_gcp_kind: str):
             futures = []
             for r in records:
-                b = json.dumps(r).encode("utf-8")
+                data = json.dumps(r).encode("utf-8")
+                compressed_data = gzip.compress(data)
+
                 try:
                     f = publisher.publish(
                         _topic_path,
-                        data=b,
+                        data=compressed_data,
                         observe_gcp_kind=observe_gcp_kind,
                         observe_cloud_function_version=_version,
+                        observe_gzip=True
```

Comment:
can we make this `observe_content_encoding=gzip` instead. That way we can handle it similar to `content-encoding` headers and support additional mechanisms if needed.

Reply:
good point, i wasn't sure about the api but content_encoding is understandable